webapp/channels/src/components/confirm_modal.tsx

```diff
@@ -166,46 +167,52 @@ export default class ConfirmModal extends React.Component<Props, State> {
             );
         }
 
+        const headerText = (
+            <div className='ConfirmModal__header'>
+                <span id='confirmModalLabel'>
+                    {this.props.title}
+                </span>
+            </div>
+        );
+
         return (
-            <Modal
+            <GenericModal
                 id={classNames('confirmModal', this.props.id)}
-                className={'modal-confirm ' + this.props.modalClass}
-                dialogClassName='a11y__modal'
+                className={'modal-confirm ' + this.props.modalClass + ' a11y__modal ConfirmModal'}
                 show={this.props.show}
                 onHide={this.handleCancel}
                 onExited={this.props.onExited}
-                role='none'
-                aria-modal={true}
-                aria-labelledby='confirmModalLabel'
-                aria-describedby='confirmModalBody'
-                data-testid={this.props.id}
+                ariaLabel='confirmModalLabel'
+                compassDesign={true}
+                modalHeaderText={headerText}
             >
-                <Modal.Header closeButton={true}>
-                    <Modal.Title
-                        componentClass='h1'
-                        id='confirmModalLabel'
-                    >
-                        {this.props.title}
-                    </Modal.Title>
-                </Modal.Header>
-                <Modal.Body id='confirmModalBody'>
-                    {this.props.message}
-                    {!this.props.checkboxInFooter && checkbox}
-                </Modal.Body>
-                <Modal.Footer>
-                    {this.props.checkboxInFooter && checkbox}
-                    {cancelButton}
-                    <button
-                        autoFocus={true}
-                        type='button'
-                        className={this.props.confirmButtonClass}
-                        onClick={this.handleConfirm}
-                        id='confirmModalButton'
+                <div
+                    aria-labelledby='confirmModalLabel'
+                    aria-describedby='confirmModalBody'
```

Comment:
Are you sure these are needed? I wouldn't expect to see them referring to children of the div that they're on